### PR TITLE
Resolve serviceAccounts on Deployments and Workspaces

### PR DESCRIPTION
**Pull request by ryanahamilton**
Our current schema has `serviceAccounts` fields on [`Deployments`](url) and [`Workspace`](https://github.com/astronomer/houston-api/blob/main/src/schema.graphql#L154) types, but they don't actually return anything when queried. This resolves that. This will allow for some query enhancements from the UI (to help astronomer/issues#1474).

### CODE REVIEW

File: src/resolvers/types/deployment/index.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import {
   generateNextTag
 } from "deployments/config";
 import { extractVariables } from "deployments/environment-variables";
+import deploymentServiceAccounts from "resolvers/query/deployment-service-accounts";
```
Comment (schnie) [inline src/resolvers/types/deployment/index.js:8]: @ryanahamilton what do you think about moving the bulk of the code that is getting imported here under the `lib` directory?

Obviously this works just fine, but importing resolvers into other resolvers is a pattern I don't think we have yet. I think of the `lib` folder as housing anything that is shared across multiple other areas/entrypoints of the app, such as other lib code, or shared across resovlers, routes or migrations (or the new workers).

I could see this kind of thing popping up elsewhere, so just want to get a good established pattern that we/others can copy/paste.

This mental model helps me keep everything straight in my head. Most things under `src` represent the different "entrypoints" of the app. These entrypoints (routes/resolvers/migrations/workers), have their own code and pull in shared code from the `lib`. What do you think?

Comment (ryanahamilton) [reply src/resolvers/types/deployment/index.js:8]: Yeah, I meant to pose that question in the PR. I'm on board with moving the code to `lib`.

One question it raises for me is how it impacts testing/coverage. It seems like the bulk of the testing should live alongside the code in `lib`, then each implementation ends up having tests as well, becoming somewhat redundant with each other and the `lib` tests. I suppose the context tests should just focus more on just confirming a response and the `lib` tests should focus on vetting the function's logic/cases?

Comment (schnie) [reply src/resolvers/types/deployment/index.js:8]: Yea I think that sounds right. Might need to switch to use the prisma client library, as opposed to the prisma bindings too.

File: src/resolvers/query/deployment-service-accounts/index.js [hunk 2]
```diff
@@ -14,24 +13,5 @@ export default async function deploymentServiceAccounts(
...
-
-  // Run final query
-  const serviceAccounts = await ctx.db.query.serviceAccounts(
-    query,
-    addFragmentToInfo(info, fragment)
-  );
-
-  // If we made it here, return the service accounts.
-  return serviceAccounts;
+  return deploymentSAs(parent, args, ctx, info);
```
[the remark this answers is not in the document]
Comment (ryanahamilton) [reply src/resolvers/query/deployment-service-accounts/index.js:16]: This was my original inclination for naming scheme. The problem was, as in this example, the top level function in this file is already using that exact name. So it had to be something different—seemed like the resolver should retain that name and the `lib` should be something different. Open to other ideas than `deploymentSAs` if you have them?

[the remark this answers is not in the document]
Comment (schnie) [reply src/resolvers/query/deployment-service-accounts/index.js:16]: Ah, didn't see that. Yea maybe this is fine now. Only other thing I could think of was like `getDeploymentServiceAccounts` or something.

[the remark this answers is not in the document]
Comment (ryanahamilton) [reply src/resolvers/query/deployment-service-accounts/index.js]: I like that suggestion better—no guessing of what "SAa" are. Just pushed update

**Comment by schnie** [comment]
@ryanahamilton are the tests failing for some reason now?

Also could we add some tests for `getWorkspaceServiceAccounts` and `getDeploymentServiceAccounts` functions? Don't want to coverage to drop.

**Comment by ryanahamilton** [comment]
@schnie sorry, was just a linting issue.

Those functions/files already have 100% coverage from their usage instances. The only dropage is coming from this unrelated file—something I keep seeing pop up.

image

**Pull request merged**